src/core/utils.ts

```diff
@@ -71,7 +71,17 @@ function isBrowserSupportedForPush() {
     const browser = getBrowserName();
 
     if ('safari' === browser) {
-        coreDependencies.push(typeof window.safari?.pushNotification);
+        if ('PushManager' in window) {
+            coreDependencies.push(
```

Comment:
here can reduce duplication a bit if do `browser safari and pushManager not in window`